libethcore/EthashGPUMiner.cpp

```diff
@@ -85,14 +91,33 @@ class EthashCLHook: public ethash_cl_miner::search_hook
 //		std::cerr << "Searched " << _count << " from " << _startNonce << std::endl;
 		m_owner->accumulateHashes(_count);
 		m_last = _startNonce + _count;
-		if (m_abort || m_owner->shouldStop())
-			return (m_aborted = true);
-		return false;
+		//if (m_abort || m_owner->shouldStop())
+		//	return (m_aborted = true);
+		//return false;
+		// TODO: still need a way to stop the GPU thread
```

Comment:
Back to the problem, I think the miner thread should just have a flag to stop (maybe it already has). The decision when to stop should be delegated to the network thread. E.g. if the stratum thread is not able to get the work for some time it should stop the miner thread. Does it make sense?